port/ztpf/omrosdump.c

```diff
@@ -254,24 +254,24 @@ omrdump_create(struct OMRPortLibrary *portLibrary, char *filename, char *dumpTyp
 				"unable to write ELF core dump: %s\n", errMsg);
 		return 1; /* Report failure, get outta here.      */
 	} else {
-		s->argv.rc = rename(s->argv.wkSpace, JVMfn); /* Success. Return.                     */
+		s->argv.rc = rename((const char *)s->argv.wkSpace, (const char *)JVMfn); /* Success. Return.                     */
 		if ((s->argv.rc < 0) && (errno == ENOENT)) {
 			//Current working directory may be different from target Dump Directory.
 			//In this scenario the dump is already in the target directory.
 			//Therefore re-issue the command using an absolute path name for old
 			//name.
-			targetDirectory = dirname(JVMfn);
+			targetDirectory = dirname((char *)JVMfn);
 			if (targetDirectory == NULL) {
 				portLibrary->tty_err_printf(portLibrary,
 						"Problem determining directory for %s\n", JVMfn);
 			}
 			// +2 for terminating Null and joining path seperator
 			oldPath = alloca(
-					strlen(targetDirectory) + strlen(s->argv.wkSpace) + 2);
+					strlen(targetDirectory) + strlen((const char *)s->argv.wkSpace) + 2);
 			strcpy(oldPath, targetDirectory);
 			strcat(oldPath, "/");
-			strcat(oldPath, s->argv.wkSpace);
-			s->argv.rc = rename(oldPath, JVMfn); /* Success. Return. */
+			strcat(oldPath, (const char *)s->argv.wkSpace);

```

Comment:
If I remove it I get this compilation error:
./ztpf/omrosdump.c: In function 'omrdump_create':
./ztpf/omrosdump.c:273:4: error: pointer targets in passing argument 2 of 'strcat' differ in signedness [-Werror=pointer-sign]
/ztpf/commit/opensource/include/string.h:142:21: note: expected 'const char * __restrict__' but argument is of type 'uint8_t *'

That particular structure member is in lowcore for us...  with that said we probably could have organized the types and their conversion better.  If it was a mainline platform path I would definitely do something about it, but this is z/TPF specific.